docs/proposals/memory-management.md

```diff
@@ -0,0 +1,135 @@
+<!-- BEGIN MUNGE: UNVERSIONED_WARNING -->
+
+<!-- BEGIN STRIP_FOR_RELEASE -->
+
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+
+<h2>PLEASE NOTE: This document applies to the HEAD of the source tree</h2>
+
+If you are using a released version of Kubernetes, you should
+refer to the docs that go with that version.
+
+<strong>
+The latest 1.0.x release of this document can be found
+[here](http://releases.k8s.io/release-1.0/docs/proposals/resource-qos.md).
+
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+# Memory Management in Kubernetes
+
+**Author**: Vishnu Kannan (@vishh)
+
+**Status**: Draft proposal; 
+
+*This document captures the current state of memory handling in kubernetes nodes and suggests some features for improving node reliability and performance*
+
+####Current Model
+Kubelet (via cAdvisor) exposes the total memory capacity on the node to master. System daemons like docker, kube-proxy, and kubelet are moved to a static cgroup. Most of the other system services, excepting init, are moved to a `/system` cgroup. Either no resource restrictions are in place for these system level cgroups or even if they do, they are set to high values (docker is limited to 70% of the total capacity). Hence it is difficult to predict the amount of memory available for pods on any given node.
```

Comment:
Are you asking whether it's Kubelet or Salt?

https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/container_manager_linux.go#L129